universal-application-tool-0.0.1/app/views/BaseHtmlView.java

```diff
@@ -78,4 +89,31 @@ protected String renderDateTime(Instant time) {
     LocalDateTime datetime = LocalDateTime.ofInstant(time, ZoneId.of("America/Los_Angeles"));
     return datetime.format(DateTimeFormatter.ofPattern("yyyy/MM/dd 'at' h:mm a"));
   }
+
+  protected ImmutableList<DomContent> createLinksAndEscapeText(String content) {
+    List<Url> urls = new UrlDetector(content, UrlDetectorOptions.Default).detect();
+    ImmutableList.Builder<DomContent> contentBuilder = ImmutableList.builder();
+    for (Url url : urls) {
+      int index = content.indexOf(url.getOriginalUrl());
+      if (index == -1) {
+        LOG.error(
+            String.format(
+                "Detected URL %s not present in actual content, %s.",
+                url.getOriginalUrl(), content));
+        continue;
+      }
+      if (index > 0) {
+        contentBuilder.add(text(content.substring(0, index)));
+      }
```

Comment:
question: It's hard for me to tell exactly what's going on here. What is index and what substring are we trying to get?

Reply:
I added a comment.